### DISCUSSION TITLE
Modifying a selector

### DISCUSSION THREAD

**Discussion by arunmmanoharan** [facebookexperimental/Recoil#1281] [Q&A] [answered]
Hi, I am fetching an API, destructuring it and storing values into individual selectors. I want to update a certain selector. I am able to do it using 

      const connection = useRecoilValue(connectionState);
      const setConnection = useSetRecoilState(connectionState);

Is there an alternate to achieve this or this the best way?

This is my codesandbox: https://codesandbox.io/s/patient-star-wc430?file=/src/Connection.jsx:167-280

**Comment by drarmstr** [comment, marked as answer]
The two hooks could be combined with `useRecoilState()`:
```jsx
const [connection, setConnection] = useRecoilState(connectionState);
```
